Userland/Libraries/LibELF/DynamicLoader.cpp

```diff
@@ -360,6 +336,48 @@ void DynamicLoader::load_program_headers()
         VERIFY_NOT_REACHED();
     }
 
+    Vector<ProgramHeaderRegion> load_regions;
+    Vector<ProgramHeaderRegion> map_regions;
+    Vector<ProgramHeaderRegion> copy_regions;
```

Comment:
to avoid memory allocs here further, would it make sense to use some inline capacity? maybe... 4 or 8 each? Not sure how many loadable/mappable/copy regions a typical binary has.

Reply:
I'm not sure if it would be worth the additional complexity, since this is after our "critical" section and they get deallocated at the end of the function anyways. What I can look into though is whether the region count is predictable.